app/View/Elements/javascript.ctp

```diff
@@ -657,4 +657,8 @@
   var defaultConfirmOk = "<?php print _txt('op.ok'); ?>";
   var defaultConfirmCancel = "<?php print _txt('op.cancel'); ?>";
   var defaultConfirmTitle = "<?php print _txt('op.confirm'); ?>";
+  
+  // Define default text for session timeout and unknown errors
+  var defaultHttp500 = "<?php print _txt('er.500'); ?>";
+  var defaultHttp401 = "<?php print _txt('er.timeout'); ?>";
```

Comment:
Note the inconsistent use of error tags here, why isn't the second one `er.http.401`?

Reply:
For the plugin we may combine two messages - one that's generic ("Unauthorized (401)" with one that's specific "Your session has timed out. Please log in to continue." Or we can make the generic 401 contain both phrases: "Unauthorized (401). Your session has timed out. Please log in to continue." Would this (or something like this) be appropriate for all 401s?

Reply:
I'm not sure, however, how useful the HTTP error code is to the end user. I think just leaving the end user message as 'er.timeout' is probably better. A developer can look under the hood to see the code returned. 

er.http.500 is different: it's a server error of some kind that just needs to be reported. I had originally told the end user "Unknown error" - but I think being more explicit here is good. (It's now "Internal server error (500)").

But I feel like an end user message for 401 should probably remain er.timeout: it's simple to understand and involves no tech speak in the end-user widget.

Reply:
I was considering HTTP 404 as a counter example, but lots of places don't report the status code to end users anymore:

- https://www.amazon.com/askdfnnn
- https://www.nytimes.com/asdfslkjalskfjslkdfjslkj

Devs just look at the network response.

Reply:
This has me thinking that "Unknown error" (or something more explicit) may still be better for a 500. Something like "The server threw an unknown error." may be better than reporting the HTTP status code to the end user. Or again, we could just make the error message a little more explicit: 

- "HTTP Error 500: The server threw an unknown error."

If we compromise on this, we could do something similar for 401:

- "HTTP Error 401: Your session has expired. Please login to continue."

If we want a compromise, how's that?

Reply:
The two examples above, if used, would probably map to the er.http.500 and er.http.401 lang strings. 

But I'm still somewhat inclined to just skip the error codes altogether, and have er.unknown and er.timeout -- Or 'er.http.500.unknown' and 'er.http.401.timeout' to make the lang strings themselves more explicit about what they represent, but the strings would just be 

- "The server threw an unknown error."
- "Your session has expired. Please login to continue."

Thoughts?